hash.go

```diff
@@ -374,8 +381,9 @@ func (h *evpHash) BlockSize() int {
 func (h *evpHash) Sum(in []byte) []byte {
 	h.init()
 	out := make([]byte, h.Size(), maxHashSize) // explicit cap to allow stack allocation
-	if C.go_hash_sum(h.ctx, h.ctx2, (*C.uchar)(unsafe.SliceData(out))) != 1 {
-		panic(newOpenSSLError("go_hash_sum"))
+	var _err *C.mkcgo_err_state
+	if C.go_hash_sum(h.ctx, h.ctx2, (*C.uchar)(unsafe.SliceData(out)), &_err) != 1 {
+		panic(newMkcgoErr("EVP_DigestFinal_ex", _err))
```

Comment:
`errst` to match what `newMkcgoErr` calls it? `_` prefix seems unusual in hand-written Go.

```suggestion
	var errst *C.mkcgo_err_state
	if C.go_hash_sum(h.ctx, h.ctx2, (*C.uchar)(unsafe.SliceData(out)), &errst) != 1 {
		panic(newMkcgoErr("EVP_DigestFinal_ex", errst))
```